src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -303,4 +338,55 @@ public function addResource(ResourceInterface $resource)
             $this->resources[$key] = $resource;
         }
     }
+
+    /**
+     * Sets an alias for an existing route.
+     *
+     * @param string       $alias The alias to create
+     * @param string|Alias $route The route to alias
+     *
+     * @throws InvalidArgumentException if the id is not a string or an Alias
+     * @throws InvalidArgumentException if the alias is for itself
+     */
+    public function addAlias(string $alias, $route): Alias
+    {
+        if (\is_string($route)) {
+            $route = new Alias($alias, $route);
+        } elseif (!$route instanceof Alias) {
+            throw new InvalidArgumentException(sprintf('Expected a string or an instance of "%s", but got "%s".', Alias::class, \is_object($route) ? \get_class($route) : \gettype($route)));
```

Comment:
```suggestion
            throw new \TypeError(sprintf('Expected a string or an instance of "%s", but got "%s".', Alias::class, get_debug_type($route)));
```

This will allow us to upgrade the signature with a union type one fine day.

Reply:
Non needed anymore.